cpp/src/lance/io/reader.cc

```diff
@@ -261,12 +261,27 @@ ::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> FileReader::ReadBatch(
   return ::arrow::RecordBatch::Make(schema.ToArrow(), arrs[0]->length(), arrs);
 }
 
-::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> FileReader::ReadBatch(
-    int32_t offset, int32_t length, const std::vector<std::string>& columns) const {
-  auto schema = manifest_->schema();
-  ARROW_ASSIGN_OR_RAISE(auto projection, schema.Project(columns));
+::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> FileReader::ReadChunk(
+    const lance::format::Schema& schema, int32_t chunk_id, std::optional<int32_t> length) const {
+  return ReadChunk(schema, chunk_id, GetArrayParams(0, length));
+}
+
+::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> FileReader::ReadChunk(
+    const lance::format::Schema& schema,
+    int32_t chunk_id,
+    std::shared_ptr<::arrow::UInt64Array> indices) const {
+  return ReadChunk(schema, chunk_id, GetArrayParams(indices));
+}
 
-  return ReadBatch(offset, length, *projection);
+::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> FileReader::ReadChunk(
+    const lance::format::Schema& schema, int32_t chunk_id, const GetArrayParams& params) const {
+  std::vector<std::shared_ptr<::arrow::Array>> arrs;
+  /// TODO: Read field in parallel.
```

Comment:
Convert this into a GH issue? feels like an easy win for a new contributor?

Reply:
Done. #43